projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -377,8 +377,11 @@ private function prepare_menu_item_url( $url, $parent_slug = '' ) {
 				return wp_specialchars_decode( esc_url_raw( $url ) );
 			}
 
-			// Disallow other external URLs.
-			return '';
+			if ( 1 !== preg_match( '/^' . preg_quote( get_site_url(), '/' ) . '/i', $url ) ) {
+				// Disallow other external URLs.
+				return '';
+			}
+			// The URL matches that of the site, treat it as an internal URL.
```

Comment:
What do you think about checking if the URL points to WP Admin as well to make it safer?

I'd also use `strpos ` instead of a regex check (for consistency with the checks above, and for readability).

```suggestion
			// Disallow other external URLs that don't point to the admin area.
			if ( 0 !== strpos( $url, get_admin_url() ) ) {
				return '';
			}
			// The URL is an admin URL, treat it as internal.
```

Reply:
> What do you think about checking if the URL points to WP Admin as well to make it safer?

I've no strong opinion. I'm not sure what it makes it safer from though, and I wonder if there were use-cases I never thought of, like if a plugin changes the front-end and needs special preview links to them for example.

> I'd also use strpos instead of a regex check (for consistency with the checks above, and for readability).

+1 that is clearer

Reply:
> I've no strong opinion. I'm not sure what it makes it safer from though, and I wonder if there were use-cases I never thought of, like if a plugin changes the front-end and needs special preview links to them for example.

I think the logic for internal URLs above check anyway it's an admin URL so I don't think there is a case where a frontend URL would be allowed anyway.

Reply:
I temporarily added a test case for `menu_item_url_data`:
```php
array(
	'/made-up-front-end-page',
	'',
	null,
	'',
),
```

and it failed because it expected an empty string and received `/made-up-front-end-page`